storage/replica_raftstorage.go

```diff
@@ -287,12 +293,14 @@ func (r *Replica) Snapshot() (raftpb.Snapshot, error) {
 		// state of the Replica). Everything must come from the snapshot.
 		snapData, err := snapshot(snap, rangeID, r.mu.state.Desc.StartKey)
 		if err != nil {
-			log.Errorf(context.TODO(), "%s: error generating snapshot: %s", r, err)
+			log.Errorf(context.Background(), "%s: error generating snapshot: %s", r, err)
```

Comment:
I know you can't thread a real context. Extracting a local variable would do little more than avoid some allocations. That there's an async task here doesn't matter - this is a non-expiring context.